src/devices/PiJuice/PiJuiceStatus.cs

```diff
@@ -0,0 +1,275 @@
+            status.PowerInput5vIo = (PowerInState)(response[0] >> 6 & 0x03);
+
+            return status;
+        }
+
+        /// <summary>
+        /// Get current fault status of PiJuice
+        /// </summary>
+        /// <returns>PiJuice fault status</returns>
+        public FaultStatus GetFaultStatus()
+        {
+            FaultStatus faultStatus = new FaultStatus();
+
+            var response = _piJuice.ReadCommand(PiJuiceCommand.FaultEvent, 1);
+
+            faultStatus.ButtonPowerOff = (response[0] & 0x01) == 0x01;
+            faultStatus.ForcedPowerOff = (response[0] & 0x02) == 0x02;
+            faultStatus.ForcedSystemPowerOff = (response[0] & 0x04) == 0x04;
+            faultStatus.WatchdogReset = (response[0] & 0x08) == 0x08;
+            faultStatus.BatteryProfileInvalid = (response[0] & 0x20) == 0x20;
+            faultStatus.BatteryChargingTempFault = (BatteryChargingTempFault)(response[0] >> 6 & 0x03);
+
+            return faultStatus;
+        }
+
+        /// <summary>
+        /// Gets event generated by PiJuice buttons presses
+        /// </summary>
+        /// <returns>List of button event types</returns>
+        public List<ButtonEventType> GetButtonEvents()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.ButtonEvent, 2);
+
+            var buttonEvents = new List<ButtonEventType>(3)
+            {
+                GetButtonEventType(response[0] & 0x0F),
+                GetButtonEventType((response[0] >> 4) & 0x0F),
+                GetButtonEventType(response[1] & 0x0F)
+            };
+
+            return buttonEvents;
+        }
+
+        /// <summary>
+        /// Clears generated button event
+        /// </summary>
+        /// <param name="button">Button to clear button event for</param>
+        public void ClearButtonEvent(Button button)
+        {
+            var array = button switch
+            {
+                Button.SW1 => new byte[] { 0xF0, 0xFF },
+                Button.SW2 => new byte[] { 0x0F, 0xFF },
+                Button.SW3 => new byte[] { 0xFF, 0xF0 },
+                _ => throw new NotImplementedException()
+            };
+
+            _piJuice.WriteCommand(PiJuiceCommand.ButtonEvent, array);
+        }
+
+        /// <summary>
+        /// Get battery charge level between 0 and 100 percent
+        /// </summary>
+        /// <returns>Battery charge level percentage</returns>
+        public byte GetChargeLevel()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.ChargeLevel, 1);
+
+            return response[0];
+        }
+
+        /// <summary>
+        /// Get battery temperature
+        /// </summary>
+        /// <returns>Battery temperature in celsius</returns>
+        public Temperature GetBatteryTemperature()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryTemperature, 2);
+
+            return new Temperature((response[1] << 8) | response[0], TemperatureUnit.DegreeCelsius);
```

Comment:
BinaryPrimitives for the conversion?

Reply:
Have you an example of using BinaryPrimitives as I've never used the class before?